src/Hosting/Hosting/src/Host.cs

```diff
@@ -70,9 +70,11 @@ public static IHostBuilder CreateDefaultBuilder(string[] args)
             builder.ConfigureAppConfiguration((hostingContext, config) =>
             {
                 var env = hostingContext.HostingEnvironment;
+                
+                var reloadOnChange = hostingContext.Configuration.GetValue("hostBuilder:reloadConfigOnChange", defaultValue: true);
```

Comment:
Should this also be set to https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationSource.cs#L33?

Apparently this condition is true: https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs#L34 even with `"hostBuilder": {"reloadConfigOnChange": false }` in appsettings.json.

kernel implementation without inotify support throw the following exception on application startup and there seems to be no way to turn (this quite an <ins>un</ins>important) feature off:

```
Unhandled exception. System.IO.IOException: Function not implemented
   at System.IO.FileSystemWatcher.StartRaisingEvents()
   at System.IO.FileSystemWatcher.StartRaisingEventsIfNotDisposed()
   at System.IO.FileSystemWatcher.set_EnableRaisingEvents(Boolean value)
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.TryEnableFileSystemWatcher()
   at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.CreateFileChangeToken(String filter)
   at Microsoft.Extensions.FileProviders.PhysicalFileProvider.Watch(String filter)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.<.ctor>b__1_0()
   at Microsoft.Extensions.Primitives.ChangeToken.OnChange(Func`1 changeTokenProducer, Action changeTokenConsumer)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider..ctor(FileConfigurationSource source)
   at Microsoft.Extensions.Configuration.Json.JsonConfigurationSource.Build(IConfigurationBuilder builder)
   at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
   at Microsoft.Extensions.Hosting.HostBuilder.BuildAppConfiguration()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
   at mymvc1.Program.Main(String[] args)
```

Reply:
got the global opt-out knob working with: https://github.com/dotnet/runtime/commit/457ed114aebe9798fd58bbf9733194e51b734896; `DOTNET_USE_POLLING_FILE_WATCHER=1` in unikernel environment completely disables inotify. sadly, it wasn't backported to net5 so i switched to net6..

Reply:
At least for HostBuilder, at the time this was being changed if you used CreateDefaultBuilder() you were getting live reloading whether you wanted it or not because it was [hard coded to true](https://github.com/dotnet/extensions/pull/2940/files/f0d48a3cae6fabe0712bf0601e1b7b312d3d3b5e#diff-0c3da3ec1ae00a375bab50fb4a1a452d745345c234877033508add5b21ffb7e1L74-R77)

If Source.ReloadOnChange is populated by an env var and is available by the time ConfigureAppConfiguration() fires then that would work a bit better.